packages/do-not-merge/src/do-not-merge.ts

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* eslint-disable node/no-extraneous-import */
+
+import {Application, Context} from 'probot';
+import Webhooks from '@octokit/webhooks';
+import {logger} from 'gcf-utils';
+
+const DO_NOT_MERGE = 'do not merge';
+const CHECK_NAME = 'Do Not Merge';
+
+const FAILURE_OUTPUT = {
+  title: 'Remove the do not merge label before merging',
+  summary: 'Remove the do not merge label before merging',
+};
+
+const SUCCESS_OUTPUT = {
+  title: 'OK to merge, label not found',
+  summary: 'OK to merge, label not found',
+};
+
+export = (app: Application) => {
+  app.on(
+    [
+      'pull_request.labeled',
+      'pull_request.unlabeled',
+      'pull_request.synchronize', // To run the check on every commit.
+    ],
+    async (
+      context: Context<Webhooks.EventPayloads.WebhookPayloadPullRequest>
+    ) => {
+      const owner = context.payload.repository.owner.login;
+      const repo = context.payload.repository.name;
+      const sha = context.payload.pull_request.head.sha;
```

Comment:
Before any of the real logic, make sure to exit if the PR is currently closed.  I think `pull_request.head` may be null if the branch gets deleted, which sometimes happens right after the merge.

Reply:
Done. Good catch!